pkg/server/plugin/upstreamauthority/spire/spire_node.go

```diff
@@ -15,20 +15,33 @@ import (
 	"github.com/spiffe/spire/pkg/common/idutil"
 	"github.com/spiffe/spire/pkg/common/util"
 	"github.com/spiffe/spire/proto/spire/api/node"
+	"github.com/spiffe/spire/proto/spire/common"
 )
 
-func submitCSRUpstreamCA(ctx context.Context, nodeClient node.NodeClient, csr []byte) ([]*x509.Certificate, *bundleutil.Bundle, error) {
-	resp, err := nodeClient.FetchX509CASVID(ctx, &node.FetchX509CASVIDRequest{
+func (m *Plugin) submitCSRUpstreamCA(ctx context.Context, csr []byte) ([]*x509.Certificate, error) {
+	m.bundleMtx.Lock()
```

Comment:
Do we want separate mutex for updating X.509 roots v.s. JWT keys?

Reply:
I thought about that... My only concern was that having too many mutexes in the struct could have a negative impact in readability/maintainability... but perhaps it's not that bad

Reply:
I think it's fine with a single mutex considering the current layout.